crates/router/src/core/payments.rs

```diff
@@ -4275,6 +4281,7 @@ pub async fn call_create_connector_customer_if_required<F, Req, D>(
     key_store: &domain::MerchantKeyStore,
     merchant_connector_account: &helpers::MerchantConnectorAccountType,
     payment_data: &mut D,
+    payment_router_data: &RouterData<F, Req, router_types::PaymentsResponseData>,
```

Comment:
I think it will be better to accept `access_token` here over RouterData. We don't want to tightly couple this function with payments flow.
```suggestion
    access_token: &AccessToken,
```